### ISSUE TITLE
Error for public test case lab 5.2, Incorrect serve event time for customer 10

### ISSUE THREAD

**Issue by hello-world-cell** [nus-cs2030/2223-s2#784]
Hello, not very sure where the error is coming from here. 
this is my output: 
<img width="341" alt="image" src="https://user-images.githubusercontent.com/67591893/230579087-013718db-3e7e-4905-bc4d-430dfc959295.png">

and this is the expected output : 
<img width="309" alt="image" src="https://user-images.githubusercontent.com/67591893/230579187-a184f8f4-57f3-445b-bb62-e5b099c26c33.png">

**Comment by trwstin** [comment]
check how you're updating the list of servers if a server rests

**Comment by NwaterM** [comment]
seems like server 1 is resting before serving customer 10...  my implementation added rest time in the done event and updates the serverlist with the new rest time too.

**Comment by hello-world-cell** [comment]
> seems like server 1 is resting before serving customer 10... my implementation added rest time in the done event and updates the serverlist with the new rest time too.

hi, thank you for your reply, but i also do update the serverlist with rest time in the done event

**Comment by edmundnwl** [comment]
besides checking if you are updating the serverlist, you should also check if you are calling the getRestTime() correctly

**Comment by notsushiii** [comment]
Perhaps you can consider checking if you called getRestTime() once only (like how we only call getServiceTime() once) and in the correct class/event.

**Comment by allnotavailable** [comment]
Your server 1 is supposed to rest before serving customer 10. You can check if you only call restTime for your human servers and not for all servers.

**Comment by easypzlemonsqueezy** [comment]
yup your server 1 is supposed to rest, might be something to do with the sequence you're calling your resttimes

**Comment by tjdbryan** [comment]
Hi, if you're confident that your serverList is being updated correctly, then you might be using the old Server when you pass it as an input from event to event. Always use the most updated Server by getting from serverList.

**Comment by vVicksv** [comment]
Since it only happend towards the end, try and check if the servers were updated correctly after executing the done event, and also that you called getRestTime() only once. It may have messed up the expected restTime of that particular server.

**Comment by eeynix** [comment]
since server 1 has to rest before serving customer 10, check if you called getRestTime() correctly when updating the list of servers in the done event
